### ISSUE TITLE
Training protoxt- adversarial training

### ISSUE THREAD

**Issue by rajasekharponakala** [BVLC/caffe#6715]
Hi Team,

I followed a convention to find the adversarial examples in the googlenet model.

For example, deploy.prototxt looks like:

force_backward: true

layer{
  name: "data"
  type: "Input"
  top: "data"
  input_param { shape: { dim: 1 dim: 1 dim: 224 dim: 224 } }
}
layer {
  name: "label"
  type: "Input"
  top: "label"
  input_param { shape: { dim: 1 dim: 1 dim: 1 dim: 1 } }
}
layer {
  name: "conv1/7x7_s2"
  type: "Convolution"
  bottom: "data"
  top: "conv1/7x7_s2"
  param {
    lr_mult: 1.0
    decay_mult: 1.0
  }
  param {
    lr_mult: 2.0
    decay_mult: 0.0
  }
.......
.......
.......
layer {
  bottom: "loss3/classifier"
  bottom: "label"
  top: "loss"
  name: "loss"
  type: "SoftmaxWithLoss"
}
 ----
This convention (Fast Gradient Sign Method) is usually finding the gradient of the loss function with respect to the image, and that loss function will have a prediction and a label.
Then the googlenet model tries to mispredict the label.
Can we compute the such adversarial examples for each minibatch in the training phase then update to the network weights. Am I missing something?

Thanks.
